tests/processors/data_contributions/test_processor.py

```diff
@@ -0,0 +1,125 @@
+        processor.process({}, conn)
+        updated_contrib = conn['explorer']['data_contributions'].find_one(id=valid_data_contrib)
+        document = conn['database']['documents'].find_one(id=updated_contrib['document_id'])
+
+        updated_contrib.update({'approved': False})
+        conn['explorer']['data_contributions'].update(updated_contrib, ['id'])
+        processor.process({}, conn)
+
+        assert conn['database']['documents'].find_one(id=document['id']) is None
+
+
+    def test_updates_contribution_document_id_if_document_is_removed(self, conn, valid_data_contrib):
+        processor.process({}, conn)
+        processed_contrib = conn['explorer']['data_contributions'].find_one(id=valid_data_contrib)
+
+        processed_contrib.update({'approved': False})
+        conn['explorer']['data_contributions'].update(processed_contrib, ['id'])
+        processor.process({}, conn)
+        updated_contrib = conn['explorer']['data_contributions'].find_one(id=valid_data_contrib)
+
+        assert processed_contrib['document_id'] is not None
+        assert updated_contrib['document_id'] is None
+
+
+    def test_ignores_contribution_with_invalid_url(self, conn, valid_data_contrib):
+        contrib = conn['explorer']['data_contributions'].find_one(id=valid_data_contrib)
+        contrib_attrs = {
+            'url': 'invalid url',
+        }
+        contrib.update(contrib_attrs)
+        conn['explorer']['data_contributions'].update(contrib, ['id'])
+        processor.process({}, conn)
+
+        updated_contrib = conn['explorer']['data_contributions'].find_one(id=valid_data_contrib)
+        assert updated_contrib['document_id'] is None
+
+
+    def test_ignores_archive_contribution(self, conn, valid_data_contrib):
+        contrib = conn['explorer']['data_contributions'].find_one(id=valid_data_contrib)
+        contrib_attrs = {
+            'url': 'https://github.com/opentrials/processors/archive/master.zip',
+        }
+        contrib.update(contrib_attrs)
+        conn['explorer']['data_contributions'].update(contrib, ['id'])
+        processor.process({}, conn)
+
+        updated_contrib = conn['explorer']['data_contributions'].find_one(id=valid_data_contrib)
+        assert updated_contrib['document_id'] is None
+
+
+    def test_ignores_contribution_with_invalid_trial_id(self, conn, valid_data_contrib):
+        contrib = conn['explorer']['data_contributions'].find_one(id=valid_data_contrib)
+        contrib_attrs = {
+            'trial_id': 'e0f3ef3c-edba-11e6-ae32-e4b3181a2c8c',
+        }
+        contrib.update(contrib_attrs)
+        conn['explorer']['data_contributions'].update(contrib, ['id'])
+        processor.process({}, conn)
+
+        updated_contrib = conn['explorer']['data_contributions'].find_one(id=valid_data_contrib)
+        assert updated_contrib['document_id'] is None
+
+
+@pytest.fixture
+def valid_data_contrib(conn, data_contribution, document_category, trial):
```

Comment:
As it looked like the same `data_contribution` fixture, only with an actual `data_category`, I removed it and added the `data_category` to the regular `data_contribution` fixture.